plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -480,6 +487,120 @@ def rpm_search_dicts(self):
                 ret.append(unit_key)
         return ret
 
+    @staticmethod
+    def _check_packages(existing_packages, new_packages):
+        """
+        Check if the new packages are the same as the existing ones.
+
+        :param existing_packages: list of packages presented in the existing erratum
+        :type  existing_packages: list of dicts
+
+        :param new_packages: list of packages presented in the new erratum
+        :type  new_packages: list of dicts
+
+        :return: True, if the lists of packages are equal
+        :rtype: bool
+        """
+        if len(existing_packages) == len(new_packages):
+            existing_packages.sort(key=itemgetter('filename'))
+            new_packages.sort(key=itemgetter('filename'))
+            return all(x == y for x, y in zip(existing_packages, new_packages))
+        return False
+
+    def merge_errata(self, other):
+        """
+        Merge two errata with the same errata_id.
+
+        There are two parts:
+        - merging of the pkglists in case of the erratum with the same id in different repositories
+        - overwriting the erratum metadata based on the `updated` field
+
+        NOTE: The second part should be eliminated after we change the way erratum is stored in the
+        MongoDB.
+
+        :param other: The erratum we are combining with the existing one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: the erratum after merge
```

Comment:
I suggest not returning anything.

The standard library takes a deliberate approach to emphasizing mutable vs. immutable types by choosing what gets returned by their methods. For example:

``'foo'.upper()`` returns a string, because it creates a new string. It cannot modify the original, because of course it is immutable. 

``[].extend([1, 2])`` does not return anything, because it does not create anything. The method has a side-effect, and the most explicit way to make that clear is to not return the original object.

Here is an email from Guido explaining his take on it: https://mail.python.org/pipermail/python-dev/2003-October/038855.html